pkg/cfn/builder/managed_nodegroup_test.go

```diff
@@ -0,0 +1,95 @@
+package builder
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/awslabs/goformation/v4"
+	"github.com/stretchr/testify/assert"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+)
+
+func TestManagedResources(t *testing.T) {
+	iamRoleTests := []struct {
+		addons                  api.NodeGroupIAMAddonPolicies
+		attachPolicyARNs        []string
+		expectedManagedPolicies []string
+	}{
+		{
+			expectedManagedPolicies: []string{"AmazonEKSWorkerNodePolicy", "AmazonEKS_CNI_Policy", "AmazonEC2ContainerRegistryReadOnly"},
+		},
+		{
+			addons: api.NodeGroupIAMAddonPolicies{
+				ImageBuilder: api.Enabled(),
+			},
+			expectedManagedPolicies: []string{"AmazonEKSWorkerNodePolicy", "AmazonEKS_CNI_Policy", "AmazonEC2ContainerRegistryReadOnly", "AmazonEC2ContainerRegistryPowerUser"},
+		},
+		{
+			addons: api.NodeGroupIAMAddonPolicies{
+				CloudWatch: api.Enabled(),
+			},
+			expectedManagedPolicies: []string{"AmazonEKSWorkerNodePolicy", "AmazonEKS_CNI_Policy", "AmazonEC2ContainerRegistryReadOnly", "CloudWatchAgentServerPolicy"},
+		},
+		{
+			attachPolicyARNs:        []string{"AmazonEKSWorkerNodePolicy", "AmazonEKS_CNI_Policy"},
+			expectedManagedPolicies: []string{"AmazonEKSWorkerNodePolicy", "AmazonEKS_CNI_Policy"},
+		},
+		// should not attach any additional policies
+		{
+			attachPolicyARNs:        []string{"CloudWatchAgentServerPolicy"},
+			expectedManagedPolicies: []string{"CloudWatchAgentServerPolicy"},
+		},
+		// no duplicate values
+		{
+			attachPolicyARNs: []string{"AmazonEC2ContainerRegistryPowerUser"},
+			addons: api.NodeGroupIAMAddonPolicies{
+				ImageBuilder: api.Enabled(),
+			},
+			expectedManagedPolicies: []string{"AmazonEC2ContainerRegistryPowerUser"},
+		},
+		{
+			attachPolicyARNs: []string{"CloudWatchAgentServerPolicy", "AmazonEC2ContainerRegistryPowerUser"},
+			addons: api.NodeGroupIAMAddonPolicies{
+				ImageBuilder: api.Enabled(),
+				CloudWatch:   api.Enabled(),
+			},
+			expectedManagedPolicies: []string{"CloudWatchAgentServerPolicy", "AmazonEC2ContainerRegistryPowerUser"},
+		},
+	}
+
+	for i, tt := range iamRoleTests {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
```

Comment:
Can we have tests descriptions instead of numbers?

Reply:
Setting the description for every test case would result in some boilerplate code (and also copy-paste as some test cases somewhat address the same case). If a test case fails, the index would tell us which one failed. Moreover, the assertions itself would include some information about the failure.

Reply:
Can you add a little text explaining what it is testing and the number as well?

Reply:
I added a little description to each test. This is how it looks like:

```
--- PASS: TestManagedResources (0.01s)
    --- PASS: TestManagedResources/0:_Default_policies (0.00s)
    --- PASS: TestManagedResources/1:_ImageBuilder_enabled (0.00s)
    --- PASS: TestManagedResources/2:_CloudWatch_enabled (0.00s)
    --- PASS: TestManagedResources/3:_Custom_policies (0.00s)
    --- PASS: TestManagedResources/4:_Custom_policies (0.00s)
    --- PASS: TestManagedResources/5:_Duplicate_policies (0.00s)
    --- PASS: TestManagedResources/6:_Multiple_duplicate_policies (0.00s)
```

Reply:
Thanks